android/KMEA/app/src/test/java/com/tavultesoft/kmea/view/FunctionalTestHelper.java

```diff
@@ -0,0 +1,76 @@
+package com.tavultesoft.kmea.view;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+import android.util.Log;
+
+import androidx.test.core.app.ApplicationProvider;
+
+import com.tavultesoft.kmea.KMManager;
+import com.tavultesoft.kmea.packages.PackageProcessor;
+import com.tavultesoft.kmea.util.FileUtils;
+
+import org.json.JSONException;
+import org.junit.Assert;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Helper methods for functional tests.
+ */
+class FunctionalTestHelper {
+
+  /**
+   * initialize keyman for tests.
+   */
+  static void initializeKeyman()
+  {
+    KMManager.setTestMode();
+    KMManager.initialize(
+      ApplicationProvider.getApplicationContext(), KMManager.KeyboardType.KEYBOARD_TYPE_INAPP);
+  }
+
+  /**
+   * set initial keyboard English.
+   */
+  static void setInitialKeyboard()
+  {
+    HashMap<String, String> keyboardInfo = KMManager.getKeyboardInfo(
+      ApplicationProvider.getApplicationContext(), 0);
+    if (keyboardInfo != null) {
+      String pkgId = keyboardInfo.get(KMManager.KMKey_PackageID);
+      String kbId = keyboardInfo.get(KMManager.KMKey_KeyboardID);
+      String langId = keyboardInfo.get(KMManager.KMKey_LanguageID);
+      String kbName = keyboardInfo.get(KMManager.KMKey_KeyboardName);
+      String langName = keyboardInfo.get(KMManager.KMKey_LanguageName);
+      String kFont = keyboardInfo.get(KMManager.KMKey_Font);
+      String kOskFont = keyboardInfo.get(KMManager.KMKey_OskFont);
+      KMManager.setKeyboard(pkgId, kbId, langId, kbName, langName, kFont, kOskFont);
```

Comment:
If you're just setting the keyboard to the default SIL_Euro_Latin keyboard, can't you just have?
```
      KMManager.setKeyboard(KMManager.KMDefault_UndefinedPackageID, 
        KMManager.KMDefault_KeyboardID,
        KMManager.KMDefault_LanguageID, 
        KMManager.KMDefault_KeyboardName,
        KMManager.KMDefault_LanguageName, 
        KMManager.KMDefault_KeyboardFont, null);
```